sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteTables.java

```diff
@@ -139,7 +139,7 @@ public void processElement(ProcessContext c) throws Exception {
         tableReference.setProjectId(
             c.getPipelineOptions().as(BigQueryOptions.class).getProject());
         tableDestination = new TableDestination(
-            tableReference, tableDestination.getTableDescription());
+            tableReference, tableDestination.getTableDescription(), tableDestination.getTimePartitioning());
```

Comment:
Thank you for the catch! Seems this bug is also present in `CreateTables.java` . Perhaps extract this logic into `tableDestination.withTableReference(...)` so that if more fields are added into TableDestination, we don't forget about them too?